ambry-api/src/main/java/com/github/ambry/quota/RequestCost.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2020 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.quota;
+
+import com.github.ambry.quota.storage.QuotaOperation;
+import java.util.HashMap;
+import java.util.Map;
+
+
+/**
+ * Class representing collection of costs associated with metrics of interest of a request for a resource.
+ */
+public class RequestCost {
+  private final Map<QuotaMetric, Double> costMap;
```

Comment:
What I don't get about this class is whether this is a per request or not? Given by it's name and the fields `quotaResource` and `quotaOperation`, it seems like every request would have its own `RequestCost`. Based on that, here is my question:

1. Why there is a `costMap`, whose key is QuotaMetric?
2. What is the difference between `QuotaMetric` and `QuotaResourceType` here.